### PR TITLE
feat: GCP Principal Credential Authentication tester

### PR DESCRIPTION
**Pull request by ravi-shankar-sap**
**Description**

Changes proposed in this pull request:

- Updated periodic http client renewal to include Authentication Check
- Report metrics to Prometheus
- Report metrics to Prometheus on all GCP calls.

**Related issue(s)**
https://jira.tools.sap/browse/PHX-110

### CODE REVIEW

File: pkg/kcp/provider/gcp/client/provider.go [hunk 1]
```diff
@@ -115,5 +119,41 @@ func newHttpClient(ctx context.Context, saJsonKeyPath string) (*http.Client, err
...
 	}
+	CheckGcpAuthentication(ctx, client, saJsonKeyPath)
 	return client, nil
 }
+
+func CheckGcpAuthentication(ctx context.Context, client *http.Client, saJsonKeyPath string) {
+	logger := composed.LoggerFromCtx(ctx)
+	cred, err := transport.Creds(ctx, option.WithCredentialsFile(saJsonKeyPath))
+	if err != nil {
+		IncrementCallCounter("Authentication", "Check", "", err)
```
Comment (abalaie) [inline pkg/kcp/provider/gcp/client/provider.go:130]: Why are we calling IncrementCallCounter only if err != nil? What is the difference between this and other cases where we call IncrementCallCounter regardless of err value?

File: pkg/kcp/provider/gcp/client/provider.go [hunk 2]
```diff
@@ -115,5 +119,41 @@ func newHttpClient(ctx context.Context, saJsonKeyPath string) (*http.Client, err
...
+		logger.Error(err, "GCP Authentication Check - error obtaining GCP credentials")
+		return
+	}
+	crmService, err := cloudresourcemanager.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		IncrementCallCounter("Authentication", "Check", "", err)
```
Comment (abalaie) [inline pkg/kcp/provider/gcp/client/provider.go:136]: Why are we calling IncrementCallCounter only if err != nil? What is the difference between this and other cases where we call IncrementCallCounter regardless of err value?

Comment (ravi-shankar-sap) [reply pkg/kcp/provider/gcp/client/provider.go:136]: Amir, these errors are from intermediate steps. i.e. they could come when the parsing JSON file failed (or) some issue with creating the GCP CRM Service. I thought we don't need to track the success of intermediate steps as they are part of the checking authentication process. If you look at the line 141, IncrementCallCounter is called even if err != nil.

**Review by abalaie — APPROVED** [review]
lgtm

**Pull request merged**